### ISSUE TITLE
Custom CA support for Tiller and AppRepository controller

### ISSUE THREAD

**Issue by obeyler** [kubeapps/kubeapps#515]
Hy, 
I deploy a chartmuseum helm repository with https and self-signed certificate.
Does it be possible to add ca on kubeapps ?
best regards

**Comment by prydonius** [comment]
Kubeapps doesn't currently support it, but it's a feature we could certainly add. How do you use your chartmuseum repo with the Helm CLI today? Are you using the `--ca-file` flag with `helm install/fetch`?

One potential way we could support this is, similar to asking for the Authorization header when adding the repo to Kubeapps, we could have a textbox for optionally entering the CA bundle. This could then be stored in a Secret and loaded by the Tiller Proxy when fetching the chart. Does that sound like it makes sense?

**Comment by obeyler** [comment]
I deploy chartmuseum without --ca-file. I expose it with an ingress rule and I've got a HaProxy in front of all my kubernetes worker then a Go Router from Cloud foundry to expose it on https
the CA certificate of https is a custom CA

**Comment by prydonius** [comment]
@obeyler and what about on the client side when using the `helm` CLI? Is the custom CA already in your local certs folder, or do you use the `--ca-file` flag?

**Comment by obeyler** [comment]
I don't use --ca-file flag on helm client.
When my CI use it deploy, it is not behind proxy so doesn't need the CA

**Comment by prydonius** [comment]
@obeyler makes sense, thanks! If ChartMuseum is running inside the same cluster as Kubeapps, you can also use the service DNS name (e.g. chartmuseum.default.svc.cluster.local:PORT) in Kubeapps to connect to it, bypassing TLS for now.

**Comment by obeyler** [comment]
The DNS name works to list the chart present into the repo but for user  who want to use the command line it offers a bad information as the helm add repo with svc.cluster.local will failed
(sorry to not anwser before I haven't seen your question before)

**Comment by prydonius** [comment]
@obeyler are you thinking about Monocular here? Kubeapps doesn't show you a `helm repo add` command.

**Comment by spiegela** [comment]
@prydonius I also am running into issues with this. I have chartmuseum installed as part of [Harbor](https://github.com/goharbor/harbor) (outside of Kubernetes) with a cert from a private CA.  Helm and Tiller appear to pick up the system trust chain, but the `apprepo-sync-...` jobs are failing with certificate errors.

Could we add a custom CA for the `apprepo-sync` containers?

**Comment by prydonius** [comment]
Thanks @spiegela, I think adding an option when adding an App Repo to specify a custom CA is worth doing.

**Comment by prydonius** [comment]
I started to test a ChartMuseum instance with SSL using instructions from https://coreos.com/os/docs/latest/generate-self-signed-certificates.html to generate a self-signed certificate. Note that the commands in that tutorial are a little out-of-date, the `-profile` option in `cfssl gencert` should be `www` and not `server`.

I ran ChartMuseum as a container and mounted the generated certs and set these options on the server: https://github.com/helm/chartmuseum#https.

The next step would be to look into how to modify the http.Request we make in chart-repo and tiller-proxy to use a custom CA that is stored in the auth secret for the app repository. After that, we can modify the frontend to accept a new input for a custom CA cert and store it in the auth Secret.
